signals/webhook/webhook.go

```diff
@@ -38,27 +37,67 @@ const (
 
 // Note: micro requires stateless operation so the Listen() method should not use the
 // receive struct to save or modify state.
-type webhook struct{}
+// webhooks are special in that they should only have one http Server since the port is fixed at runtime
+// this means that webhook signals are stateful, however losing this state is not a concern since
+// the connections will be re-initialized by the SignalClient
+type webhook struct {
+	srv *http.Server
+	sync.RWMutex
+	inactiveEndpoints map[string]struct{}
+}
+
+// New creates a new webhook listener for the specified port
+func New(port int) sdk.Listener {
+	srv := &http.Server{
+		Addr: fmt.Sprintf(":%v", port),
+		// Good practice to enforce timeouts to avoid Slowloris attacks
+		WriteTimeout: time.Second * 5,
+		ReadTimeout:  time.Second * 5,
+		IdleTimeout:  time.Second * 30,
+	}
+	// Start http server
+	go func() {
+		log.Printf("starting http server listening on: %s", srv.Addr)
+		err := srv.ListenAndServe()
```

Comment:
In a follow-up commit, we should support TLS.

Reply:
good idea

Reply:
I'll allow supporting TLS in next commit

Reply:
If this is expected to run inside a kubernetes pod, what is the best place to get the cert and key file and/or certificates?

Reply:
I'm going to push this off, if this is really needed, let's raise another issue for this

Reply:
Let's do another issue for it. Two possibilities:

1. Self-signed cert (the webhook signal pod will simply create it's own cert and private key)
2. User provided cert and key (maybe volume mounted in the webhook signal pod?)